### PR TITLE
If a workflow user rejects an item that they submitted, the item becomes uneditable

### LINKED ISSUE #8042: If a workflow user rejects an item that they submitted, the item becomes uneditable
**Describe the bug**

If a certain users (who is also a reviewer) submits an item and they then reject their own item in the workflow, the item cannot be edited any longer in the workspace.

This is because all WRITE rights have been removed from that item.

**To Reproduce**
1. Create (a non admin) user whom is both submitter in a certain collection and a reviewer
2. Submit an item in that collection
3. Reject that item yourself
4. Go back to your workspace & attempt to modify some metadata & save
5. This won't work & you will get a message stating: "There was an issue when saving the item, please try again later."

**Expected behavior**
There is prob something wrong either with the granting or removing permissions. The permissions should be granted / removed with the proper "type" ( ResourcePolicy.TYPE_SUBMISSION/ ResourcePolicy.TYPE_WORKFLOW). There is prob something wrong with a check somewhere.

### PR DESCRIPTION
**Pull request by KevinVdV**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #8042

## Description
If a certain users (who is also a reviewer) submits an item and they then reject their own item in the workflow, the item cannot be edited any longer in the workspace.

This is because all WRITE rights have been removed from that item.

## Instructions for Reviewers

* Create (a non admin) user whom is both submitter in a certain collection and a reviewer
* Submit an item in that collection
* Reject that item yourself
* Go back to your workspace & attempt to modify some metadata & save
* This won't work & you will get a message stating: "There was an issue when saving the item, please try again later."

List of changes in this PR:
* We moved the removal of the RIGHTS before we create the new workspace item (which in turn creates new RIGHTS for the submitter user.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [X] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [X] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [X] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [X] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [X] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [X] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-api/src/test/java/org/dspace/xmlworkflow/XmlWorkflowServiceIT.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        Community community = CommunityBuilder.createCommunity(context)
+                                              .withName("Parent Community")
+                                              .build();
+        Collection colWithWorkflow = CollectionBuilder.createCollection(context, community)
+                                                      .withName("Collection WITH workflow")
+                                                      .withWorkflowGroup(1, submitter)
+                                                      .build();
+        Workflow workflow = XmlWorkflowServiceFactory.getInstance().getWorkflowFactory().getWorkflow(colWithWorkflow);
+        ClaimedTask taskToReject = ClaimedTaskBuilder.createClaimedTask(context, colWithWorkflow, submitter)
+                                                     .withTitle("Test workflow item to reject").build();
```
Comment (tdonohue) [inline dspace-api/src/test/java/org/dspace/xmlworkflow/XmlWorkflowServiceIT.java:71]: @KevinVdV : You look to be leaving the authorization turned **off** for the entirety of the test...that can result in odd/unexpected behaviors.  You should ensure you are turning the auth system back *on* immediately after this line of code, as you only need it turned off to create test content.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@KevinVdV or @MarieVerdonck  : Overall, the code here looks correct, but we need to ensure that the Integration Test isn't running with the authorization system **turned off**.  While I don't think this setting is impacting your test currently, it can obviously result in odd behavior in tests (as tests may pass that should otherwise fail).  If you could correct your IT, then I think this would be ready to merge.

File: dspace-api/src/test/java/org/dspace/xmlworkflow/XmlWorkflowServiceIT.java (hunk 1 above)
Comment (KevinVdV) [reply dspace-api/src/test/java/org/dspace/xmlworkflow/XmlWorkflowServiceIT.java:71]: @tdonohue You are right, I missed that part in my internal review. Quickly fixed it with a commit.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @KevinVdV & @MarieVerdonck  !  This all looks correct and makes complete sense to me.  Tests prove that it works as described.

**Pull request merged**